### PR TITLE
docs: Add requirements.txt and a few comments

### PR DESCRIPTION
**Pull request by kate-feingold**
* A leaner requirements txt
* Conditional imports of BA
* A few comments

### CODE REVIEW

**Review by kate-feingold — COMMENTED** [review]
.

File: code/models/SetOfSet.py [hunk 1]
```diff
@@ -55,8 +57,8 @@ def __init__(self, conf):
         self.m_net = get_linear_layers([num_feats] * 2 + [m_d_out], final_layer=True, batchnorm=False)
         self.n_net = get_linear_layers([num_feats] * 2 + [n_d_out], final_layer=True, batchnorm=False)
 
-    def forward(self, data):
-        x = data.x  # x is [m,n,d] sparse matrix
+    def forward(self, data: SceneData):
+        x: SparseMat = data.x  # x is [m,n,d] sparse matrix
```
Comment (kate-feingold) [inline code/models/SetOfSet.py:61]: Let's remove this

**Review by kate-feingold — COMMENTED** [review]
Hi hello, left a few comments

File: code/models/SetOfSet.py (hunk 1 above)
Comment (kate-feingold) [reply code/models/SetOfSet.py:61]: It's helpful for pycharm suggestions 
Or I can do it and mention a commit 57f41c0

**Review by D0nsh — APPROVED** [review]

**Pull request merged**